demos/solps_from_balance.py

```diff
@@ -0,0 +1,163 @@
+
+# Copyright 2014-2017 United Kingdom Atomic Energy Authority
+#
+# Licensed under the EUPL, Version 1.1 or – as soon they will be approved by the
+# European Commission - subsequent versions of the EUPL (the "Licence");
+# You may not use this work except in compliance with the Licence.
+# You may obtain a copy of the Licence at:
+#
+# https://joinup.ec.europa.eu/software/page/eupl5
+#
+# Unless required by applicable law or agreed to in writing, software distributed
+# under the Licence is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR
+# CONDITIONS OF ANY KIND, either express or implied.
+#
+# See the Licence for the specific language governing permissions and limitations
+# under the Licence.
+
+import matplotlib.pyplot as plt
+import numpy as np
+
+from cherab.core.atomic.elements import carbon, deuterium
+from cherab.solps import load_solps_from_balance
+
+plt.ion()
+
+xl, xu = (0.0, 2.0)
+yl, yu = (-2.0, 2.0)
+
+print('Enter name of balance.nc file:')
+filename = input()
```

Comment:
I like that users can specify any balance.nc file here in the demo, but you do need to be careful about the assumptions you make. For example, not all runs will have D0, D1 and C0-C6: I have some C-Mod runs with no carbon. I wouldn't worry about modifying the code to catch all these different possibilities, but I would suggest adding a comment here to say that for this demo to run properly the input file must match the assumptions you're making about its contents.